From 36140b4cf3bfcc4bcbb9f64e2d312b209196f631 Mon Sep 17 00:00:00 2001
From: Li Guohui <liguohui2061@phytium.com.cn>
Date: Thu, 16 May 2024 15:21:01 +0800
Subject: [PATCH 043/360] dt-bindings: mailbox: phytium: Add a bindings for
 Phytium mailbox

This patch documents the DT bindings for Phytium KCS interface.

Signed-off-by: Li Guohui <liguohui2061@phytium.com.cn>
Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
---
 .../bindings/mailbox/phytium,mbox.yaml        | 48 +++++++++++++++++++
 1 file changed, 48 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/mailbox/phytium,mbox.yaml

diff --git a/Documentation/devicetree/bindings/mailbox/phytium,mbox.yaml b/Documentation/devicetree/bindings/mailbox/phytium,mbox.yaml
new file mode 100644
index 000000000000..d875b08b5532
--- /dev/null
+++ b/Documentation/devicetree/bindings/mailbox/phytium,mbox.yaml
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/mailbox/phytium,mbox.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium Mailbox Driver
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+description: |
+  The Phytium mailbox controller that has a channel/link to communicate
+  with the remote end. A link raises interrupt for any received data. However,
+  there is no specified way of knowing if the sent data has been read by the
+  remote. This driver assumes the sender polls STAT register and the remote
+  clears it after having read the data.
+
+properties:
+  compatible:
+    const: phytium,mbox
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    maxItems: 1
+
+  '#mbox-cells':
+    const: 1
+    description: the index of the channel needed
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - '#mbox-cells'
+
+additionalProperties: false
+
+examples:
+  - |
+    mbox: mailbox@2a000000 {
+      compatible = "phytium,mbox";
+      reg = <0x0 0x2a000000 0x0 0x1000>;
+      #mbox-cells = <1>;
+      interrupts = <0 48 4>;
+    };
-- 
2.20.1

